src/main/java/com/mraof/minestuck/client/gui/ReadableSburbCodeScreen.java

```diff
@@ -0,0 +1,285 @@
+ */
+public class ReadableSburbCodeScreen extends Screen
+{
+	private static final ResourceLocation BOOK_TEXTURES_01A = new ResourceLocation(Minestuck.MOD_ID, "textures/gui/sburb_book_01a.png");
+	private static final ResourceLocation BOOK_TEXTURES_01B = new ResourceLocation(Minestuck.MOD_ID, "textures/gui/sburb_book_01b.png");
+	private static final ResourceLocation BOOK_TEXTURES_02 = new ResourceLocation(Minestuck.MOD_ID, "textures/gui/sburb_book_02.png");
+	private static final ResourceLocation BOOK_TEXTURES_03 = new ResourceLocation(Minestuck.MOD_ID, "textures/gui/sburb_book_03.png");
+	private final boolean[] hieroglyphValidityArray; //same size as MAX_HIEROGLYPH_COUNT
+	private final List<List<String>> listOfPages = new ArrayList<>(); //each element of the outermost list is a different page, and each page is a collection of lines
+	
+	private PageButton forwardButton;
+	private PageButton backButton;
+	
+	private List<String> textList = null;
+	private int linesPerBlock = 27; //With 298 lines of text the default is 27 per block
+	private int totalPages = 8;
+	private int currentPage = 0;
+	
+	//GUI sizes
+	private static final int GUI_WIDTH = 192;
+	private static final int GUI_HEIGHT = 192;
+	private static final int TEXT_WIDTH = 288; //114
+	private static final int TEXT_OFFSET_X = 36, TEXT_OFFSET_Y = 32;
+	
+	private static final int CUSTOM_LINE_HEIGHT = 3;
+	private static final int LINES_PER_PAGE = 40; //how many lines can be fit neatly on a page, at 48 character per line
+	
+	private static final List<Block> FULL_HIEROGLYPH_LIST = MSTags.getBlocksFromTag(MSTags.Blocks.GREEN_HIEROGLYPHS);
+	private static final int MAX_HIEROGLYPH_COUNT = FULL_HIEROGLYPH_LIST.size() + 1;
+	
+	private final static String EMPTY_SPACE = "                                                "; //48 characters
+	
+	public ReadableSburbCodeScreen(List<Block> recordedBlockList, boolean paradoxCode)
+	{
+		super(NarratorChatListener.NO_TITLE);
+		this.hieroglyphValidityArray = checkForValidity(recordedBlockList, paradoxCode);
+		
+	}
+	
+	@Override
+	protected void init()
+	{
+		super.init();
+		this.createMenuControls();
+		this.createPageControlButtons();
+		
+		try
+		{
+			Resource resource = this.minecraft.getResourceManager().getResource(new ResourceLocation(Minestuck.MOD_ID, "texts/rana_temporaria_sec22b.txt")); //The text is broken into lines 48 characters long to neatly fit within a block of text that avoids empty spaces
+			InputStream inputStream = resource.getInputStream();
+			BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
+			textList = bufferedReader.lines().collect(Collectors.toList());
+			
+			linesPerBlock = textList.size() / MAX_HIEROGLYPH_COUNT;
+			totalPages = Math.round((float) (textList.size() + 1) / (float) LINES_PER_PAGE);
+		} catch(Exception ignored)
+		{
+		}
```

Comment:
You should ad a finally clause that tries to close the resource in the case of an error (see `WinScreen.wrapCreditsIO()`. You may as well also log the exception if any occur)